### PR TITLE
Add required/optional marks for builders

### LINKED ISSUE #183: Write about Optional and Required values in all Builders
For example: 

```java
/**
* Required: Specifies table name
* <p/>
* Default value is <code>null</code>
*
* @param table non-null table name
* @return builder
*/
@NonNull
public Builder table(@NonNull String table) {
  this.table = table;
  return this;
}

/**
* Optional: Specifies arguments for "where" clause
* <p/>
* Passed objects will be immediately converted to {@link String} via calling {@link Object#toString()}
* <p/>
* Default value is <code>null</code>
*
* @param whereArgs array of objects that will be converted to strings
* @return builder
*/
@NonNull
public Builder whereArgs(@Nullable Object... whereArgs) {
  this.whereArgs = QueryUtil.varargsToList(whereArgs);
  return this;
}
```

### PR DESCRIPTION
**Pull request by nikitin-da**
Closes #183 
@artem-zinnatullin Сheck please

### CODE REVIEW

File: storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 /**
  * Insert query for {@link com.pushtorefresh.storio.contentresolver.StorIOContentResolver}
- * <p>
+ * <p/>
```
Comment (artem-zinnatullin) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java]: What is wrong with IntelliJ IDEA..

File: storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java [hunk 2]
```diff
@@ -71,9 +71,9 @@ public Builder uri(@NonNull Uri uri) {
 
         /**
          * Builds {@link InsertQuery} instance with required params,
-         * can throw {@link IllegalStateException} if something is incorrect
+         * can throw {@link NullPointerException} if one of required params wasn't specify
```
Comment (artem-zinnatullin) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java]: I think, we can remove information about exceptions, do you agree?

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java [hunk 3]
```diff
@@ -175,22 +175,46 @@ public void endTransaction() {
...
 
         private SQLiteDatabase db;
 
+        /**
+         * Specifies database for internal usage.
+         * You should provide this or {@link SQLiteOpenHelper}
+         * @see {@link #sqliteOpenHelper(SQLiteOpenHelper)}
+         *
+         * @param db a real database for internal usage
+         */
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java]: Missing `return` javadoc

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java [hunk 4]
```diff
@@ -175,22 +175,46 @@ public void endTransaction() {
...
             return this;
         }
 
+        /**
+         * Specifies SqLite helper for internal usage
+         * You should provide this or {@link SQLiteDatabase}
+         * @see {@link #db(SQLiteDatabase)}
+         *
+         * @param sqliteOpenHelper a SqLite helper for internal usage
+         */
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java]: Missing javadoc

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java [hunk 5]
```diff
@@ -175,22 +175,46 @@ public void endTransaction() {
...
         @NonNull
         public Builder sqliteOpenHelper(@NonNull SQLiteOpenHelper sqliteOpenHelper) {
             db = sqliteOpenHelper.getWritableDatabase();
             return this;
         }
 
+        /**
+         * Builds {@link DefaultStorIOSQLiteDb} instance with required params,
+         * can throw {@link NullPointerException} if required parameter missed
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java]: Same question about exception

File: storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java (hunk 1 above)
Comment (nikitin-da) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java]: Is it Idea bug? Are there any differences with these tags. 
I see that `<p>` doesn't work when there isn't tag section in javadoc.
But works "better" in another cases =(

File: storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java (hunk 2 above)
Comment (nikitin-da) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java]: Absolutely

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java (hunk 3 above)
Comment (nikitin-da) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/DefaultStorIOSQLiteDb.java]: thanks

File: storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java (hunk 1 above)
Comment (artem-zinnatullin) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/query/InsertQuery.java]: I'll Google it later

**Comment by nikitin-da** [comment]
Updated

**Comment by artem-zinnatullin** [comment]
:+1: Thanks!

**Pull request merged**